protostar/starknet/cheaters/contracts.py

```diff
@@ -263,3 +262,40 @@ def _transform_data_to_cairo_format(
             raise ConstructorInputTransformationException(
                 f"There was an error while parsing constructor arguments:\n{dt_exc.message}",
             ) from dt_exc
+
+    async def get_contract_class_at(self, contract_address: int) -> ContractClass:
+        class_hash_bytes = await self.cheatable_state.get_class_hash_at(
+            int(contract_address)
+        )
+        return await self.cheatable_state.get_contract_class(class_hash_bytes)
+
+    async def invoke(
+        self,
+        function_name: str,
+        contract_address: Address,
+        calldata: Optional[CairoOrPythonData] = None,
+    ):
+        contract_address_int = int(contract_address)
+        if isinstance(calldata, collections.Mapping):
+            contract_class = await self.get_contract_class_at(contract_address_int)
+            assert (
+                contract_class.abi
+            ), f"No abi found for the contract at {contract_address}"
+
+            transformer = from_python_transformer(
+                contract_class.abi, function_name, "inputs"
+            )
+            cairo_calldata = transformer(calldata)
+        else:
+            cairo_calldata = calldata or []
```

Comment:
I think this should be moved to a separate function, I did this in #1358 so I'd wait for that PR to be merged and hen rebase, what do you think?

Reply:
sure, we can wait for that one